File: Python/Raw/oop-python/sorting.py
//---------------------integer sort-----------------------
data = []
data.insert(0, 8)
data.insert(1, 80)
data.insert(2, 17)
data.insert(3, 4)
data.insert(4, 90)
print("Before Sorting")
for i in data:
    print(i)

print("After Sorting")
for i in sorted(data):
    print(i)

//------------------------Float Sort-------------------------------

doubleList = [3.2, 2.14, 43.24, 66.23, 1.2, 5.6, 23.1, 34.2, 123]
print("Before Sorting")
for i in doubleList:
    print(i)

print("After Sorting")
for i in sorted(doubleList):
    print(i)

//--------------------------String Sort-------------------------------------------

stringList = ["Monkey", "Donkey", "Wolverine", "Elephant", "Lion", "Lamb"]
print("Before Sorting")
for i in stringList:
    print(i)

print("After Sorting")
for i in sorted(stringList):
    print(i)

File: JavaScript/Raw/oop-js/sorting.js
//---------------------integer sort-----------------------
let data = [];
data[0] = 8;
data[1] = 80;
data[2] = 17;
data[3] = 4;
data[4] = 90;
console.log("Before Sorting");

for(var i = 0; i < data.length; i++){
    console.log(data[i]);
}
console.log("After Sorting");

var numArray = data.sort((a,b) => a-b);
// function compareFunction(a, b){
//     // 1. if < 0 ... a
//     // 2. 0 ... nothing will be change
//     //3. if > 0 ... b
//     return a - b;
// }
// data.sort(compareFunction);
// console.log(data);
console.log(numArray);
//------------------------Float Sort-------------------------------
let doubleList = [3.2, 2.14, 43.24, 66.23, 1.2, 5.6, 23.1, 34.2, 123];
console.log("Before Sorting");

for(var i = 0; i < doubleList.length; i++){
    console.log(doubleList[i]);
}
console.log("After Sorting");

let doubleArray = doubleList.sort((a,b) => a-b);
console.log(doubleArray);

//--------------------------String Sort-------------------------------------------

let stringList = ["Monkey", "Donkey", "Wolverine", "Elephant", "Lion", "Lamb"];
console.log("Before Sorting");

for(var i = 0; i < stringList.length; i++){
    console.log(stringList[i]);
}
console.log("After Sorting");
let stringArray = stringList.sort();
for(var i = 0; i < stringArray.length; i++){
    console.log(stringArray[i]);
}